mono/metadata/Makefile.am

```diff
@@ -90,6 +90,43 @@ if !ENABLE_ILGEN
 ilgen_libraries = libmono-ilgen.la
 endif
 
+if ENABLE_MONOTOUCH
+platform_sources += ../../support/zlib-helper.c
```

Comment:
we probably need a `platform_c_sources` and some CFLAGS stuff to make sure it's compiled without the `CXX_CFLAGS` when building Mono with `--enable-cxx`  --- `support/` is not going to be ported to C++.